src/wrappers/python/PyOpenEXR.cpp

```diff
@@ -0,0 +1,2755 @@
+                                               sizeof (unsigned int) * 1,       // xStride
+                                               sizeof (unsigned int) * width)); // yStride
+
+    std::vector<std::shared_ptr<Array2DVoidPtr>> sliceDatas;
+
+    for (auto c : channels)
+    {
+        const PyChannel& C = c.second.cast<const PyChannel&>();
+
+        if (C.pixels.dtype().kind() != 'O')
+            throw std::runtime_error("Expected deep pixel array with dtype 'O'");
+
+        C._type = NUM_PIXELTYPES;
+        
+        int nrgba = get_deep_nrgba(C.pixels);
+        if (nrgba == 0)
+            C.insertDeepSlice(frameBuffer, C.name, height, width, nrgba, dw_offset, 0, sampleCount, sliceDatas);
+        else
+        {
+            std::string name_prefix = "";
+            if (C.name != "RGB" && C.name != "RGBA")
+                name_prefix = C.name + ".";
+
+            C.insertDeepSlice(frameBuffer, name_prefix+"R", height, width, nrgba, dw_offset, 0, sampleCount, sliceDatas);
+            C.insertDeepSlice(frameBuffer, name_prefix+"G", height, width, nrgba, dw_offset, 1, sampleCount, sliceDatas);
+            C.insertDeepSlice(frameBuffer, name_prefix+"B", height, width, nrgba, dw_offset, 2, sampleCount, sliceDatas);
+            if (nrgba == 4)
+                C.insertDeepSlice(frameBuffer, name_prefix+"A", height, width, nrgba, dw_offset, 3, sampleCount, sliceDatas);
+        }
+    }
+
+    if (type() == EXR_STORAGE_DEEP_SCANLINE)
+    {
+        DeepScanLineOutputPart part(outfile, part_index);
+        part.setFrameBuffer (frameBuffer);
+        part.writePixels (height);
+    }
+    else 
+    {
+        DeepTiledOutputPart part(outfile, part_index);
+        part.setFrameBuffer (frameBuffer);
+
+        for (int y = 0; y < part.numYTiles (0); y++)
+            for (int x = 0; x < part.numXTiles (0); x++)
+                part.writeTile (x, y, 0);
+    }
+}
+
+//
+// Return whether "name" corresponds to one of the 'R', 'G', 'B', or 'A'
+// channels in a "RGBA" tuple of channels. Return 4 if there's an 'A'
+// channel, 3 if it's just RGB, and 0 otherwise.
+//
+// py_channel_name is returned as either the prefix, e.g. "left" for
+// "left.R", "left.G", "left.B", or "RGBA" if the channel names are just 'R',
+// 'G', and 'B'.
+//
+// This means:
+//
+//     channels["left"] = np.array((height,width,3))
+// or:
+//     channels["RGB"] = np.array((height,width,3))
+//
+// channel_name is returned as the single character name of the channel
+//
+
+int
+PyPart::rgbaChannel(const ChannelList& channel_list, const std::string& name,
+                    std::string& py_channel_name, char& channel_name)
+{
+    py_channel_name = name;
+    channel_name = py_channel_name.back();
+    if (channel_name == 'R' ||
+        channel_name == 'G' ||
+        channel_name == 'B' ||
+        channel_name == 'A')
+    {
+        // has the right final character. The preceding character is either a
```

Comment:
Truncated comment. 

Also for this entire method -- is it appropriate to write it on the python side instead of the C side? It seems just some string manipulations and would be expressed more naturally in python, without much performance penalty of moving it off C code.

Reply:
The function is used throughout this file, moving it would mean you'd have to move most of the implementation to Python....

Reply:
Good point... Let's leave it then (but finish the comment :)